modules/sdk-core/src/bitgo/utils/tss/baseTypes.ts

```diff
@@ -0,0 +1,120 @@
+import { SerializedKeyPair } from 'openpgp';
+import { IRequestTracer } from '../../../api';
+import { KeychainsTriplet } from '../../baseCoin';
+import { Keychain } from '../../keychain';
+import { Memo, WalletType } from '../../wallet/iWallet';
+
+export type TxRequestVersion = 'full' | 'lite';
+
+export interface PrebuildTransactionWithIntentOptions {
+  reqId: IRequestTracer;
+  intentType: string;
+  sequenceId?: string;
+  recipients: {
+    address: string;
+    amount: string | number;
+    tokenName?: string;
+  }[];
+  comment?: string;
+  memo?: Memo;
+  tokenName?: string;
+  nonce?: string;
+}
+
+export type TxRequestState =
+  | 'pendingApproval'
+  | 'canceled'
+  | 'rejected'
+  | 'initialized'
+  | 'pendingDelivery'
+  | 'delivered'
+  | 'pendingUserSignature'
+  | 'signed';
+
+export type TransactionState =
+  | 'initialized'
+  | 'pendingSignature'
+  | 'signed'
+  | 'held'
+  | 'delivered'
+  | 'invalidSignature'
+  | 'rejected';
+
+export type ITxRequest = {
+  txRequestId: string;
+  walletId: string;
+  walletType: WalletType;
+  version: number;
+  enterpriseId?: string;
+  state: TxRequestState;
+  date: string;
+  userId: string;
+  intent: any;
+  pendingApprovalId?: string;
+  policiesChecked: boolean;
+  signatureShares?: SignatureShareRecord[];
+  pendingTxHashes?: string[];
+  txHashes?: string[];
+  // Only available in 'lite' version
+  unsignedTxs: unknown[];
+  // Only available in 'full' version
+  transactions: {
+    state: TransactionState;
+    unsignedTx: unknown;
+    signatureShares: SignatureShareRecord[];
+  }[];
+  apiVersion?: TxRequestVersion;
+  latest: boolean;
+};
+
+export enum SignatureShareType {
+  USER = 'user',
+  BACKUP = 'backup',
+  BITGO = 'bitgo',
+}
+
+export interface SignatureShareRecord {
+  from: SignatureShareType;
+  to: SignatureShareType;
+  share: string;
+}
+
+export interface ITssUtils<KeyShare, TxRequest> {
```

Comment:
Yeah, Because for different signature schemes or blockchains, the `unsignedTxs` /  `transactions`  `intents` will be different and seems best to have it overridden at the time of initialisation

Reply:
TxRequest has been removed from parameters based on our discussion over slack